benchmarks/envoy_proxy.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+"""@package envoy_proxy.py
+
+Contains customized fixture & EnvoyProxyServer abstraction for use in tests.
+"""
+
+import logging
+import os
+import pytest
+
+from test.integration.integration_test_fixtures import (HttpIntegrationTestBase,
+                                                        determineIpVersionsFromEnvironment)
+from test.integration.nighthawk_test_server import NighthawkTestServer
+
+
+class EnvoyProxyServer(NighthawkTestServer):
+  """
+  Envoy proxy server abstraction. Note that it derives from NighthawkTestServer, as that
+  is implemented as a customized Envoy, which is convenient here: the CLI and admin interface
+  mechanics that we rely on are the same. So all we do here, is specialize so we can override
+  the docker image and binary name.
+  """
+
+  def __init__(self, config_template_path, server_ip, ip_version, parameters=dict(), tag=""):
```

Comment:
Can we add a comment for the initializer describing its arguments?

The same comment applies to all public functions, classes and methods.